### PR TITLE
Implement oracle signing algo for client side

### PR DESCRIPTION
**Pull request by benthecarman**
https://github.com/discreetlogcontracts/dlcspecs/pull/113

### CODE REVIEW

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala [hunk 1]
```diff
@@ -156,7 +156,10 @@ class DLCPaneModel(resultArea: TextArea, oracleInfoArea: TextArea) {
                                              "dummy oracle")
 
           val announcementSig =
-            privKey.schnorrSign(CryptoUtil.sha256(oracleEvent.bytes).bytes)
+            privKey.schnorrSign(
+              CryptoUtil
+                .taggedSha256(oracleEvent.bytes, "DLC/oracle/announcement/v0")
+                .bytes)
```
Comment (nkohen) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala]: all invocations of `taggedSha256` should be replaced with a new function in `CryptoUtil` akin to 
https://github.com/bitcoin-s/bitcoin-s/blob/master/crypto/src/main/scala/org/bitcoins/crypto/CryptoUtil.scala#L71
including the per-compiled val for the tag's hash.

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala [hunk 2]
```diff
@@ -172,7 +175,9 @@ class DLCPaneModel(resultArea: TextArea, oracleInfoArea: TextArea) {
               builder.append("Outcomes and oracle sigs in order of entry:\n")
               contractInfo.keys.foreach { outcome =>
                 val bytes = outcome.serialized.head
-                val hash = CryptoUtil.sha256(bytes).bytes
+                val hash = CryptoUtil
+                  .taggedSha256(bytes, "DLC/oracle/attestation/v0")
```
Comment (nkohen) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCPaneModel.scala]: ditto for attestations as well

File: dlc/src/main/scala/org/bitcoins/dlc/testgen/DLCTestUtil.scala [hunk 3]
```diff
@@ -67,7 +67,7 @@ object DLCTestUtil {
     val func = DLCPayoutCurve(
       Vector(
         OutcomePayoutEndpoint(0, leftVal),
-        OutcomePayoutEndpoint(botCollar, leftVal),
+        OutcomePayoutEndpoint(botCollar + 1, leftVal),
```
Comment (nkohen) [inline dlc/src/main/scala/org/bitcoins/dlc/testgen/DLCTestUtil.scala:80]: I assume this fix made tests pass in the `testUntilFailed` command?

**Review by nkohen — CHANGES_REQUESTED** [review]
Concept ACK

**Comment by nkohen** [comment]
Also, so you know, me and Tibo agreed that this is acceptable to get in now, we're both willing to work around this. We just need to make sure to remember to apply the diff to https://github.com/discreetlogcontracts/dlcspecs/pull/125

File: dlc/src/main/scala/org/bitcoins/dlc/testgen/DLCTestUtil.scala (hunk 3 above)
Comment (benthecarman) [reply dlc/src/main/scala/org/bitcoins/dlc/testgen/DLCTestUtil.scala:80]: Yes

**Review by nkohen — COMMENTED** [review]
ACK after you add a scaladoc to each of the `_TagBytes` vals in `CryptoUtil.scala` containing the string that was used to generate these (as, I hope, is done for the other vals like this in that file)

**Review by nkohen — APPROVED** [review]

**Pull request merged**